external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala

```diff
@@ -1497,6 +1497,32 @@ abstract class AvroSuite extends QueryTest with SharedSparkSession {
       |}
     """.stripMargin)
   }
+
+  test("log a warning of ignoreExtension deprecation") {
+    val logAppender = new LogAppender
+    withTempPath { dir =>
+      Seq(("a", 1, 2), ("b", 1, 2), ("c", 2, 1), ("d", 2, 1))
+        .toDF("value", "p1", "p2")
+        .repartition(2)
+        .write
+        .format("avro")
+        .option("header", true)
+        .save(dir.getCanonicalPath)
+      withLogAppender(logAppender) {
+        spark
+          .read
+          .format("avro")
+          .option(AvroOptions.ignoreExtensionKey, false)
+          .option("header", true)
+          .load(dir.getCanonicalPath)
+          .count()
+      }
+      val deprecatedEvents = logAppender.loggingEvents
+        .filter(_.getRenderedMessage.contains(
+          s"Option ${AvroOptions.ignoreExtensionKey} is deprecated"))
+      assert(deprecatedEvents.size === 1)
```

Comment:
Can we test if the size is just bigger then 0 just in case we have other deprecation logs later?

Reply:
I compared the size to 1 to avoid any concerns that it is printed multiple times like in the PR (https://github.com/apache/spark/pull/27174#discussion_r365596481), per each partition.

Reply:
If we expect it is printed only once, maybe we should assert that?

Reply:
So, if somebody modifies the code in the future in the way the warning is printed many times, we will catch the situation by the test.

Reply:
> just in case we have other deprecation logs later?

As we discussed in another PR, we are not going to print any log warnings about `ignoreExtension`. Am I right or misunderstood something?